### ISSUE TITLE
Run javascript from widget callback

### ISSUE THREAD

**Issue by DYSpider13** [jupyter-widgets/ipywidgets#2458]
Hello, 

Is it possible to execute a javascript inside a widget callback (a button for instance) ? If not, is there an alternative for that ? 

Best

**Comment by hainm** [comment]
I think so, using `from IPython.display import Javascript`

```
from IPython.display import display, Javascript

Javascript("console.log('hello')")
```

**Comment by SylvainCorlay** [comment]
> I think so, using from IPython.display import Javascript

Yes, although this kind of javascript injection is not allowed in jupyterlab.

> Is it possible to execute a javascript inside a widget callback (a button for instance)

We have a solution for this that was used in ipysheet using web workers to sandbox the unsafe js, although the executed javascript does not have access to the DOM of the main page or the js context of the main page.

**Comment by hainm** [comment]
> Yes, although this kind of javascript injection is not allowed in jupyterlab.

which lab version that does not allow this? It work fine with lab 0.35


![ok](https://user-images.githubusercontent.com/4451957/60213226-0d91f700-9831-11e9-9eba-d4be1bc869ee.png)

**Comment by SylvainCorlay** [comment]
OK, my information is outdated. That used to not be allowed, which seems to be a good thing to me.

**Comment by DYSpider13** [comment]
Thank you all. Using `Javascript("console.log('hello')")` did not work, I had to put it into display like this: 

`display(Javascript("console.log('hello')"))`

Any idea why ?

**Comment by SylvainCorlay** [comment]
@DYSpider13 I really do not recommand using this though. Injecting javascript is a big security red flag and may be disabled in many deployments of Jupyter.

**Comment by renefritze** [comment]
It looks like Bokeh has a solution for this [here](https://docs.bokeh.org/en/latest/docs/user_guide/interaction/callbacks.html)

Me personally I would prefer a solution closer to plain ipywidgets since I would have no actual use for Bokeh besides the CustomJS callbacks.

Somewhat related would be #1109

**Comment by jasongrout** [comment]
> Me personally I would prefer a solution closer to plain ipywidgets since I would have no actual use for Bokeh besides the CustomJS callbacks.

You can with the display Javascript trick above. That's a generic Jupyter thing, nothing to do with ipywidgets (living below the layer of ipywidgets).

To Sylvain's point - a problem with executing javascript is that it most often makes your code only work in a single frontend (and often only a single major version of that frontend), so it has limited usefulness. The approach of ipywidgets in general is to get desired functionality into the frontend through whatever the frontend extension system there is and then triggering the functionality through a message to the frontend.

**Comment by sruthiiyer** [comment]
How to substitute variable value inside code tag of HTML in ipywidgets
 

`object_fit = [A.ISONoise(always_apply=False, p=1.0, intensity=(0.1, 0.5), color_shift=(0.01, 0.05))]


h = HTML(value='<{size}><code>ISONoise(always_apply=False, p=1.0, intensity=(0.1, 0.5), color_shift=(0.01, 0.05))</code> </{size}>'.format(size='h4'))`


The value inside code keeps verying. So, how to substitute it. I'm new to HTML

**Comment by sruthiiyer** [comment]
> How to substitute variable value inside code tag of HTML in ipywidgets
> 
> `object_fit = [A.ISONoise(always_apply=False, p=1.0, intensity=(0.1, 0.5), color_shift=(0.01, 0.05))]
> 
> h = HTML(value='<{size}>`ISONoise(always_apply=False, p=1.0, intensity=(0.1, 0.5), color_shift=(0.01, 0.05))` </{size}>'.format(size='h4'))`
> 
> The value inside code keeps verying. So, how to substitute it. I'm new to HTML

I want substitute the object_fit variable inside the code tag
